pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -255,9 +255,8 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
         has_reached_max_steps = self.trainer.max_steps and self.trainer.max_steps <= global_step
 
         global_step += 1
-        if self.has_trained:
-            if not has_reached_max_steps:
-                current_epoch += 1
+        if not has_reached_max_steps:
+            current_epoch += 1
```

Comment:
n00b question: could you explain why this increments global step / current epoch?

Reply:
Because current_epoch is updated when training epoch start so for the current epoch it is saved with +1 to make sure that next time it will start training from the next epoch.